### PR TITLE
[OCI fix] Add tenancy specific prefix to zone in runtime (use general catalog file)

### PR DESCRIPTION
**Pull request by HysunHe**
This is a change made corresponding with the OCI catalog file (please kindly review the catalog file PR as well). In previous catalog file, there is a prefix in the AvailabilityZone, which is tenancy specific. So customer need to manually replace this prefix with according to his tenancy. This fix removes this neccesarity. The prefix in the catalog file is removed, and the program will find the correct prefix in runtime.

Tested (run the relevant ones):
- Sky launch / sky down
- [v] Any manual or new tests for this PR (please specify below)
- [v] All smoke tests: `pytest tests/test_smoke.py` 
- [v] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [v] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

**Review by Michaelvll — APPROVED** [review]
Thanks for fixing this @HysunHe! LGTM. We may need to ask the user to `rm -r ~/.sky/catalogs/v5/oci` after upgrading to this PR, so that there catalog can be get updated as well.

File: sky/clouds/oci.py [hunk 1]
```diff
@@ -358,6 +365,13 @@ def check_credentials(cls) -> Tuple[bool, Optional[str]]:
             del user
             # TODO[Hysun]: More privilege check can be added
             return True, None
+        except (oci_adaptor.get_oci().exceptions.ConfigFileNotFound,
+                oci_adaptor.get_oci().exceptions.InvalidConfig) as e:
+            # This should only happen in testing where oci config is
+            # monkeypatched.
```
Comment (Michaelvll) [inline sky/clouds/oci.py:371]: Is it possible that in non-testing environment, the user can have some invalid configuration that reaches here? If not, might worth commenting about why.

File: sky/clouds/oci.py [hunk 2]
```diff
@@ -231,15 +231,22 @@ def make_deploy_resources_variables(
...
+                ad_list = identity_client.list_availability_domains(
+                    compartment_id=oci_adaptor.get_oci_config(
+                        profile=oci_conf.get_profile())['tenancy']).data
+
+                first_ad = ad_list[0]
+                _tenancy_prefix = str(first_ad.name).split(':')[0]
+            except (oci_adaptor.get_oci().exceptions.ConfigFileNotFound,
+                    oci_adaptor.get_oci().exceptions.InvalidConfig) as e:
+                # This should only happen in testing where oci config is
+                # monkeypatched.
```
Comment (Michaelvll) [inline sky/clouds/oci.py:247]: ditto. It worth explaining why these errors will not happen in non-test cases.

Comment (HysunHe) [reply sky/clouds/oci.py:247]: Because in non-testing env. The sky check even cannot passed.

Comment (HysunHe) [reply sky/clouds/oci.py:247]: Let me add more comments

File: sky/clouds/oci.py [hunk 3]
```diff
@@ -227,13 +229,33 @@ def make_deploy_resources_variables(
...
+                    compartment_id=oci_adaptor.get_oci_config(
+                        profile=oci_conf.get_profile())['tenancy']).data
+
+                first_ad = ad_list[0]
+                _tenancy_prefix = str(first_ad.name).split(':')[0]
+            except (oci_adaptor.get_oci().exceptions.ConfigFileNotFound,
+                    oci_adaptor.get_oci().exceptions.InvalidConfig) as e:
+                # This should only happen in testing where oci config is
+                # monkeypatched. In real use, if the OCI config is not
+                # valid, the 'sky check' would fail(OCI disabled).
```
Comment (Michaelvll) [inline sky/clouds/oci.py:248]:
```suggestion
                # valid, the 'sky check' would fail (OCI disabled).
```

File: sky/clouds/oci.py [hunk 4]
```diff
@@ -339,19 +361,25 @@ def check_credentials(cls) -> Tuple[bool, Optional[str]]:
...
             # TODO[Hysun]: More privilege check can be added
             return True, None
-        except oci_adaptor.service_exception():
+        except (oci_adaptor.get_oci().exceptions.ConfigFileNotFound,
+                oci_adaptor.get_oci().exceptions.InvalidConfig,
+                oci_adaptor.service_exception()) as e:
+            logger.warning(f'OCI config error: {str(e)}')
             return False, (f'OCI credential is not correctly set. '
                            f'Check the credential file at {conf_file}\n'
                            f'{cls._INDENT_PREFIX}{credential_help_str}')
```
Comment (Michaelvll) [inline sky/clouds/oci.py:375]:
```suggestion
            return False, (f'OCI credential is not correctly set. '
                           f'Check the credential file at {conf_file}\n'
                           f'{cls._INDENT_PREFIX}{credential_help_str}\n'
                           f'{cls._INDENT_PREFIX}Error details: '
                           f'{common_utils.format_exception(e, use_bracket=True)}')
```

File: sky/clouds/oci.py [hunk 5]
```diff
@@ -339,19 +361,25 @@ def check_credentials(cls) -> Tuple[bool, Optional[str]]:
...
+                    oci_adaptor.get_oci_config(
+                        profile=oci_conf.get_profile())['user']).data
             del user
             # TODO[Hysun]: More privilege check can be added
             return True, None
-        except oci_adaptor.service_exception():
+        except (oci_adaptor.get_oci().exceptions.ConfigFileNotFound,
+                oci_adaptor.get_oci().exceptions.InvalidConfig,
+                oci_adaptor.service_exception()) as e:
+            logger.warning(f'OCI config error: {str(e)}')
```
[the remark this answers is not in the document]
Comment (HysunHe) [reply sky/clouds/oci.py:372]: ah, yes, my bad

[the remark this answers is not in the document]
Comment (HysunHe) [reply sky/clouds/oci.py:372]: Next time for my local CI testing, i must REMEMBER change my local valid oci config to a invalid one, always forgot it.

[the remark this answers is not in the document]
Comment (Michaelvll) [reply sky/clouds/oci.py]: No worries at all! ; ) It looks good to me now.

**Pull request merged**